tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -23,75 +23,768 @@ THE SOFTWARE.
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h2,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing Struct type, checks padding, assigning integer to a char
+    hipLaunchKernelStruct_t3 hipLaunchKernelStruct_h3;
+    hipLaunchKernelStruct_h3.bf1 = 1;
+    hipLaunchKernelStruct_h3.bf2 = 1;
+    hipLaunchKernelStruct_h3.l1 = 1.0;
+    hipLaunchKernelStruct_h3.bf3 = 1;
+    hipLaunchKernelStruct_h3.result = false;
+                // initialize to false, will be set to
+                // true if the struct size is 1Byte, from device size
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc3),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h3,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing empty struct
+    hipLaunchKernelStruct_t4 hipLaunchKernelStruct_h4;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc4),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h4,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing struct with pointer object to a hipLaunchKernelGGL()
+    hipLaunchKernelStruct_t5 hipLaunchKernelStruct_h5;
+    char* cp_d5;  // This is passed as pointer to struct member
+    // allocating memory for char pointer on device
+    HIPCHECK(hipMalloc((void**)&cp_d5, sizeof(char)));
+    HIPCHECK(hipMemset(cp_d5, 'p', sizeof(char)));
+    hipLaunchKernelStruct_h5.c1 = 'c';
+    hipLaunchKernelStruct_h5.cp = cp_d5;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc5),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h5,
+                    result_d);
+    ResultValidation();
+
+    // Test: Passing struct with aligned(8)
+    hipLaunchKernelStruct_t6 hipLaunchKernelStruct_h6;
+    hipLaunchKernelStruct_h6.c1 = 'c';
+    hipLaunchKernelStruct_h6.si = 1;
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc6),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h6,
+                    result_d);
+    // alignment is broken hence disabled the validation part
+    #if ENABLE_ALIGNMENT_TEST
+    ResultValidation();
+    #endif
+
+
+    // Test: Passing struct with aligned(16)
+    hipLaunchKernelStruct_t7 hipLaunchKernelStruct_h7;
+    hipLaunchKernelStruct_h7.c1 = 'c';
+    hipLaunchKernelStruct_h7.si = 1;
+    #if ENABLE_ALIGNMENT_TEST  // This is broken on small bar
+    hipLaunchKernelGGL(HIP_KERNEL_NAME(hipLaunchKernelStructFunc7),
+                    dim3(BLOCK_DIM_SIZE),
+                    dim3(1), 0, 0, hipLaunchKernelStruct_h7,
+                    result_d);
+    ResultValidation();
+    #endif
+
+    // Test: Passing struct with packed aligned to 6Bytes
+    hipLaunchKernelStruct_t8 hipLaunchKernelStruct_h8;
```

Comment:
can you add another test just to check packed struct size is expected? You can reuse this structure type, just remove the aligned attribute but keep the packed attribute.

Reply:
can you also add another test which has struct type argument with an aligned data member?

Reply:
Done, added two more tests as you suggested, aligned(4) & packed. 
Test names:
hipLaunchKernelStructFunc8A for packed
hipLaunchKernelStructFunc8B for aligned(4)

Results: 
Packed broken
alignment working fine.

Reply:
To be more specific about aligment member:  with latest changes, alignment is working fine on large bar setup but still it's broken on small bar hence tests disabled